packages/next/server/web/spec-extension/fetch-event.ts

```diff
@@ -10,13 +10,29 @@ export class NextFetchEvent extends FetchEvent {
     this.sourcePage = params.page
   }
 
-  // @ts-ignore
+  /**
+   * @deprecated The `request` is now the first parameter and the API is now async.
+   *
+   * Read more: https://nextjs.org/docs/messages/middleware-new-signature
+   */
   get request() {
     throw new DeprecationError({
       page: this.sourcePage,
     })
   }
 
+  /**
+   * @deprecated `respondWith` is no longer needed and the API is now async
+   * with the form:
+   *
+   * ```ts
```

Comment:
Same as https://github.com/vercel/next.js/pull/35448#discussion_r830502589 for consistency